### PR TITLE
limb disabled refactor

### LINKED ISSUE #53219: Paraplegic trait bugged, the disabled can walk
## Round ID: 
[144976](https://scrubby.melonmesa.com/round/144976) and multiple others, possibly subsequent rounds after this.

## Reproduction:
Choose the paraplegic trait and unbuckle yourself from the wheelchair, should have full range of movement and not be impaired.

The trait itself wasn't bugged prior to the recent update yesterday, which resulted in the trait breaking.

### PR DESCRIPTION
**Pull request by Rohesie**
* Fixes #53219
* Nukes `is_disabled()`
* Turns several variable value changes into events.
* `bodypart_disabled` turned into a boolean value.
* `BODYPART_DISABLED_WOUND` turned into `TRAIT_DISABLED_BY_WOUND`. Not the cleanest thing out there, but it works.
* Cleans some code and reduces number of updates, as only the extremities have a disabled effect.

### CODE REVIEW

File: code/__DEFINES/traits.dm [hunk 1]
```diff
@@ -203,6 +203,7 @@ Remember to update _globalvars/traits.dm if you're adding/removing/renaming trai
 
 //non-mob traits
 #define TRAIT_PARALYSIS				"paralysis" //Used for limb-based paralysis, where replacing the limb will fix it
+#define TRAIT_DISABLED_BY_WOUND		"disabled-by-wound" //Used for limbs.
```
Comment (Timberpoes) [inline code/__DEFINES/traits.dm:206]: DMDocify pretty please!

**Review by Timberpoes — CHANGES_REQUESTED** [review]
I can't see any glaring or obvious issues that jump out at me.

Is this all tested and working and ready to be merged?

**Comment by Timberpoes** [comment]
I do have a question just in general. It's not a code review thing per se, but more of just a general overarching question to understand things better.

Using bodypart_disabled instead of is_disabled() - How would a child inheriting from this override or otherwise alter deciding when a limb is or isn't disabled?

I'm going to assume the reason is_disabled() was a proc before was so children could inherit it or so the proc could do some sort of sanity testing to make sure the limb was truly disabled first - But I could (and might well be) wrong on that.

**Comment by Rohesie** [comment]
The idea is to remove any and all redefinition of `is_thing()` checks.
If a child gets disabled on different circumstances, then it should handle the setting of the disabled var properly in these events, so it has `bodypart_disabled` true only when it's the case. For that, it can add behavior related to `set_bodypart_disabled()`, ideally at the base of the proc in a generic way.

The reason to do that is because it's very easy and lazy to just redefine `is_thing()` checks, but after you do it and you solve your immediate problem you no longer have a way to track the event of it changing. If `is_thing()` is a complex check, instead of a variable that gets turned on and off by a chain of events, then you are simply unable to append behavior to when the variable changes.

The idea here is to move towards more events-oriented code. I say that redefined/complex `is_thing()` checks are evil and should be substituted with variables that get updated accordingly, and you'll notice I've been doing several of these refactors.

**Review by Timberpoes — APPROVED** [review]

**Pull request merged**